src/PerfView/PerfViewData.cs

```diff
@@ -4153,6 +4154,72 @@ protected internal override StackSource OpenStackSourceImpl(string streamName, T
+            {
+                var objectTypeMap = new Dictionary<long, long>(1000);
+                var interopTraceEventParser = new InteropTraceEventParser(eventSource);
+                interopTraceEventParser.AddCallbackForEvent<TaskCCWCreationArgs>(null, args =>
+                {
+                    if(!objectTypeMap.ContainsKey(args.objectID))
+                        objectTypeMap.Add(args.objectID, args.targetObjectIDType);
+                });
+                interopTraceEventParser.AddCallbackForEvents<TaskCCWQueryRuntimeClassNameArgs>(args =>
+                {
+                    sample.TimeRelativeMSec = args.TimeStampRelativeMSec;
+                    var stackIndex = stackSource.GetCallStack(args.CallStackIndex(), args);
+
+                    var ccwRefKindName = "CCW QueryRuntimeClassName";
+                    var ccwRefKindNameIndex = stackSource.Interner.FrameIntern(ccwRefKindName);
+                    stackIndex = stackSource.Interner.CallStackIntern(ccwRefKindNameIndex, stackIndex);
+
+                    sample.StackIndex = stackIndex;
+                    stackSource.AddSample(sample);
+                });
+                interopTraceEventParser.AddCallbackForEvents<TaskCCWRefCountIncArgs>(args => {
+                    sample.Metric = 1;
+                    sample.TimeRelativeMSec = args.TimeStampRelativeMSec;
+                    var stackIndex = stackSource.GetCallStack(args.CallStackIndex(), args);
+                    
+                    var ccwRefKindName = "CCW RefCountInc";
+                    var ccwRefKindNameIndex = stackSource.Interner.FrameIntern(ccwRefKindName);
+                    stackIndex = stackSource.Interner.CallStackIntern(ccwRefKindNameIndex, stackIndex);
+
+                    var objectId = "Object ID " + args.objectID;
+                    var objectIdIndex = stackSource.Interner.FrameIntern(objectId);
+                    stackIndex = stackSource.Interner.CallStackIntern(objectIdIndex, stackIndex);
+
+                    if (objectTypeMap.ContainsKey(args.objectID))
+                    {
+                        var objectType = "Object Type " + objectTypeMap[args.objectID];
+                        var objectTypeIndex = stackSource.Interner.FrameIntern(objectType);
+                        stackIndex = stackSource.Interner.CallStackIntern(objectTypeIndex, stackIndex);
+                    }
+                    var ccwRefCount = "CCW NewRefCnt " + args.refCount;
+                    var ccwRefCountIndex = stackSource.Interner.FrameIntern(ccwRefCount.ToString());
+                    stackIndex = stackSource.Interner.CallStackIntern(ccwRefCountIndex, stackIndex);
+
+                    sample.StackIndex = stackIndex;
+                    stackSource.AddSample(sample);
+                });
+                interopTraceEventParser.AddCallbackForEvents<TaskCCWRefCountDecArgs>(args =>
+                {
+                    sample.Metric = -1;
```

Comment:
This body should be identical to the TaskCCWRefCountIncArgs case except that the metric is -1.     The method above is now complex enough that you should make a helper that both of these call.   (The helper will take the timestamp, object ID and the refcount and metric as parameters).

Reply:
done